include/stdexec/execution.hpp

```diff
@@ -3466,7 +3511,96 @@ namespace stdexec {
+      }
+    };
+
+    template <class _Receiver, class _Tag, class... _Args>
+    constexpr auto __rcvr_vfun_fn(_Tag (*)(_Args...)) noexcept {
+      return +[](void* __pointer, _Args&&... __args) noexcept {
+        _Receiver* __rcvr = static_cast<_Receiver*>(__pointer);
+        _Tag{}(static_cast<_Receiver&&>(*__rcvr), static_cast<_Args&&>(__args)...);
+      };
+    }
+  } // namespace __any_
+
   namespace __let {
+
+    template <class _Sigs, class _Env>
+    struct __receiver_vtable_for;
+
+    template <class _Env, class... _Sigs>
+    struct __receiver_vtable_for<completion_signatures<_Sigs...>, _Env>
+      : stdexec::__any_::__rcvr_vfun<_Sigs>... {
+
+      _Env (*__do_get_env)(const void* __rcvr) noexcept;
+
+      template <class _Receiver>
+      explicit constexpr __receiver_vtable_for(const _Receiver*) noexcept
+        : stdexec::__any_::__rcvr_vfun<_Sigs>{stdexec::__any_::__rcvr_vfun_fn<_Receiver>(
+          (_Sigs*) nullptr)}...
+        , __do_get_env{+[](const void* __rcvr) noexcept -> _Env {
+          return stdexec::get_env(*static_cast<const _Receiver*>(__rcvr));
+        }} {
+      }
+
+      using stdexec::__any_::__rcvr_vfun<_Sigs>::operator()...;
+
+      auto __get_env(const void* __rcvr) const noexcept -> _Env {
+        return __do_get_env(__rcvr);
+      }
+    };
+
+    template <class _Receiver, class _Sigs>
+    inline constexpr __receiver_vtable_for<_Sigs, env_of_t<_Receiver>> __receiver_vtable_for_v{
+      (const _Receiver*) nullptr};
+
+    template <class _Sigs, class _Env, class _SchedulerOrNoneSuch>
+    class __receiver_ref {
+     public:
+      using receiver_concept = receiver_t;
+
+      template <class _Receiver>
+      __receiver_ref(_Receiver& __rcvr, _SchedulerOrNoneSuch __sched = {}) noexcept(
+        __nothrow_move_constructible<_SchedulerOrNoneSuch>)
+        : __vtable_{&__receiver_vtable_for_v<_Receiver, _Sigs>}
+        , __receiver_{&__rcvr}
+        , __scheduler_{static_cast<_SchedulerOrNoneSuch&&>(__sched)} {
+      }
+
+      template <same_as<get_env_t> _Tag>
+      friend auto tag_invoke(_Tag, const __receiver_ref& __self) noexcept {
+        if constexpr (same_as<_SchedulerOrNoneSuch, __none_such>) {
+          return __self.__vtable_->__get_env(__self.__receiver_);
+        } else {
+          return __env::__join(
+            __env::__with(__self.__scheduler_, get_scheduler),
+            __env::__without(__self.__vtable_->__get_env(__self.__receiver_), get_domain));
+        }
+      }
+
+      template <__completion_tag _Tag, same_as<__receiver_ref> _Self, class... _As>
+      friend void tag_invoke(_Tag, _Self&& __self, _As&&... __as) noexcept {
+        (*__self.__vtable_)(__self.__receiver_, _Tag(), static_cast<_As&&>(__as)...);
+      }
+
+     private:
+      const __receiver_vtable_for<_Sigs, _Env>* __vtable_;
+      void* __receiver_;
+      STDEXEC_ATTRIBUTE((no_unique_address))
+      _SchedulerOrNoneSuch __scheduler_;
+    };
```

Comment:
I'm a bit surprised to see all this stuff in the `__let` namespace instead of `__any_`. What's the reason?

Reply:
This receiver ref has a slightly different flavor.

1. completion functions are type erased 
2. call to get_env is type erased
3. we pass an additional scheduler

I think the third point is not well implemented and could be hidden behind the type erased call to get env. I will try to address that.

The receiver_ref in `any_sender` type-erases a given set of queries in addition to the completion functions, i.e. the environment itself is type-erased, too. I think those two `receiver_refs` are different. :thinking: 

...
Maybe something like `receiver_ref<Sigs, any_env<Queries...>>` might be possible (even a template specialization). What do you think?